include/engine/geospatial_query.hpp

```diff
@@ -457,11 +460,21 @@ template <typename RTreeT, typename DataFacadeT> class GeospatialQuery
                             forward_durations.begin() + data.fwd_segment_position,
                             EdgeDuration{0});
 
+        EdgeDistance forward_distance_offset = 0;
+        for (auto current = std::next(forward_geometry.begin());
+             current < forward_geometry.begin() + data.fwd_segment_position + 1;
+             ++current)
+        {
+            forward_distance_offset += util::coordinate_calculation::haversineDistance(
+                datafacade.GetCoordinateOfNode(*std::prev(current)),
+                datafacade.GetCoordinateOfNode(*current));
+        }
+
```

Comment:
if you agree, let's change this to

```
for (auto current = forward_geometry.begin();
   current < forward_geometry.begin() + data.fwd_segment_position;
   ++current)
{
  forward_distance_offset += util::coordinate_calculation::haversineDistance(
      datafacade.GetCoordinateOfNode(*current),
      datafacade.GetCoordinateOfNode(*std::next(current)));
}
```

i cant explain but i find it more intuitive. change if you agree :)